.github/workflows/build-template.yml

```diff
@@ -0,0 +1,45 @@
+name: Build the LaTeX template
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  VERSION_FILE: NOVAthesisFiles/nt-version.sty
+  ROW_VERSION: 1
+  START_POS_VERSION: 33
+  EMAIL_GITHUB: joao.lourenco@fct.unl.pt
+  USERNAME_GITHUB: joaomlourenco
+
+permissions: write-all
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3.3.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Compile the LaTeX template
+        uses: xu-cheng/latex-action@v2
+        with:
+          compiler: latexmk
```

Comment:
This is the default compiler.  And the flag in line 31 only works if the compiler is not changed.  I think the best is to remove this line.

Reply:
Also, the PDF I want as showcase is not the one in the default configuration.
I use the script below to generate the final PDF.   Can we use it as the "compiler" instead?  Would the action still work?

```bash
#!/usr/bin/env bash

MAKE=$(which gmake)
[[ -n "$MAKE" ]] || MAKE=make

CONF=Config/1_novathesis.tex

# save the config file
cp ${CONF} ${CONF}.OLD

# select options for showcase PDF: final phd document form FCT-NOVA, with final index (_índice remissivo_) and trimed book spine
perl -pi -e "s,.*\\\\ntsetup{doctype=.*},\\\\ntsetup{doctype=phd}," $CONF
perl -pi -e "s,.*\\\\ntsetup{school=.*},\\\\ntsetup{school=nova/fct}," $CONF
perl -pi -e "s,.*\\\\ntsetup{docstatus=.*},\\\\ntsetup{docstatus=final}," $CONF
perl -pi -e "s,.*\\\\ntsetup{spine=.*},\\\\ntsetup{spine=trim}," $CONF
perl -pi -e "s,.*\\\\ntsetup{print/index=.*},\\\\ntsetup{print/index=true}," $CONF

# build the PDF
$MAKE

# restore defaults
mv ${CONF}.OLD ${CONF}
```